cumulus/parachains/integration-tests/emulated/common/src/impls.rs

```diff
@@ -356,6 +346,38 @@ macro_rules! impl_hrmp_channels_helpers_for_relay_chain {
 	};
 }
 
+#[macro_export]
+macro_rules! impl_send_transact_helpers_for_relay_chain {
+	( $chain:ident ) => {
+		$crate::impls::paste::paste! {
+			impl $chain {
+				/// A root origin (as governance) sends `xcm::Transact` with encoded `call` to child parachain.
+				pub fn send_transact_to_parachain(
+					origin_kind: $crate::impls::OriginKind,
+					recipient: $crate::impls::ParaId,
+					call: $crate::impls::DoubleEncoded<()>
+				) {
+					use $crate::impls::{bx, Chain, RelayChain};
+
+					<Self as $crate::impls::TestExt>::execute_with(|| {
+						let root_origin = <Self as Chain>::RuntimeOrigin::root();
```

Comment:
Why are we assuming that is going to be always `root`? Someone would like to test sending a `Transact` with different `origin`. I think we should either accept `origin ` as an extra argument for `send_transact_to_parachain` or rename method to `send_transact_to_parachain_as_sudo` (better the first option)

Reply:
Well, this is meant for governance-like calls from relay to para with `RuntimeOrigin::root()` and `UnpaidExecution` not to repeat the same code everywhere:
```
/// A root origin (as governance) sends `xcm::Transact` with encoded `call` to child parachain.
```

If we want it make it more general, then yes `origin` and also some "switch" for `UnpaidExecution/BuyExecution` wrapper should be added as a parameter. Then we should add it as a generic helper function for `Chain` trait. 

I think, for me, `send_transact_to_parachain_as_sudo` means more like there is `pallet_sudo` which is not the case. 

I am inclined to keep it as it is :).
Anyway, anybody else can still create whatever scenario with `[<$chain Pallet>]>::XcmPallet::send`.

@NachoPal  wdyt?

Reply:
Didn't realised about the hardcoded `UnpaidExcution`. Are there cases where an `UnpaidExecution` is going to work with a different origin than `Superuser`? If not then I do not see it necessary to have it as function argument.

Sorry, I sometimes use `sudo` and `root` interchangeably. If it is meant to be for governance-like calls (`root`) then we can do:

```rust
pub fn send_transact_to_parachain_as_root(
					recipient: $crate::impls::ParaId,
					call: $crate::impls::DoubleEncoded<()>
				)
```

Why I am trying to avoid is to impl a method that is only meant to work for `root`, `UnpaidExecution` and `Origin::Superuser` with a so generic name - `send_transact_to_parachain` - (even if is properly documented)